runbot_travis2docker/models/runbot_build.py

```diff
@@ -183,9 +183,10 @@ def _get_ssh_keys(self):
         if not response:
             return
         keys = ""
+        login = ''
```

Comment:
This line shouldn't be needed

Reply:
Here is the exception I get in the OCA runbot logs:

```
2020-01-08 09:32:00,443 4492 ERROR ocarunbot odoo.addons.base.ir.ir_cron: Call from cron Runbot Cron for server action #187 failed in Job #11
Traceback (most recent call last):
  File "/srv/openerp/instances/runbot2-1/parts/runbot-addons/runbot_travis2docker/models/runbot_build.py", line 188, in _get_ssh_keys
    login = response.get(own_key).get('login')
AttributeError: 'NoneType' object has no attribute 'get'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/srv/openerp/instances/runbot2-1/parts/server/odoo/tools/safe_eval.py", line 350, in safe_eval
    return unsafe_eval(c, globals_dict, locals_dict)
  File "", line 1, in <module>
  File "/srv/openerp/instances/runbot2-1/parts/runbot/runbot/models/repo.py", line 327, in _cron
    self._scheduler(repos.ids)
  File "/srv/openerp/instances/runbot2-1/parts/runbot/runbot/models/repo.py", line 256, in _scheduler
    build_ids._schedule()
  File "/srv/openerp/instances/runbot2-1/parts/runbot-addons/runbot_travis2docker/models/runbot_build.py", line 200, in _schedule
    res = super(RunbotBuild, self)._schedule()
  File "/srv/openerp/instances/runbot2-1/parts/runbot/runbot/models/build.py", line 463, in _schedule
    build._schedule()
  File "/srv/openerp/instances/runbot2-1/parts/runbot-addons/runbot_travis2docker/models/runbot_build.py", line 200, in _schedule
    res = super(RunbotBuild, self)._schedule()
  File "/srv/openerp/instances/runbot2-1/parts/runbot/runbot/models/build.py", line 463, in _schedule
    build._schedule()
  File "/srv/openerp/instances/runbot2-1/parts/runbot-addons/runbot_travis2docker/models/runbot_build.py", line 211, in _schedule
    ssh_keys = build._get_ssh_keys() or ''
  File "/srv/openerp/instances/runbot2-1/parts/runbot-addons/runbot_travis2docker/models/runbot_build.py", line 195, in _get_ssh_keys
    "Error fetching %s (%s): %s", own_key, login, err)
UnboundLocalError: local variable 'login' referenced before assignment

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/srv/openerp/instances/runbot2-1/parts/server/odoo/addons/base/ir/ir_cron.py", line 102, in _callback
    self.env['ir.actions.server'].browse(server_action_id).run()
  File "/srv/openerp/instances/runbot2-1/parts/server/odoo/addons/base/ir/ir_actions.py", line 557, in run
    res = func(action, eval_context=eval_context)
  File "/srv/openerp/instances/runbot2-1/parts/server/addons/website/models/ir_actions.py", line 57, in run_action_code_multi
    res = super(ServerAction, self).run_action_code_multi(action, eval_context)
  File "/srv/openerp/instances/runbot2-1/parts/server/odoo/addons/base/ir/ir_actions.py", line 433, in run_action_code_multi
    safe_eval(action.sudo().code.strip(), eval_context, mode="exec", nocopy=True)  # nocopy allows to return 'action'
  File "/srv/openerp/instances/runbot2-1/parts/server/odoo/tools/safe_eval.py", line 373, in safe_eval
    pycompat.reraise(ValueError, ValueError('%s: "%s" while evaluating\n%r' % (ustr(type(e)), ustr(e), expr)), exc_info[2])
  File "/srv/openerp/instances/runbot2-1/parts/server/odoo/tools/pycompat.py", line 86, in reraise
    raise value.with_traceback(tb)
  File "/srv/openerp/instances/runbot2-1/parts/server/odoo/tools/safe_eval.py", line 350, in safe_eval
    return unsafe_eval(c, globals_dict, locals_dict)
  File "", line 1, in <module>
  File "/srv/openerp/instances/runbot2-1/parts/runbot/runbot/models/repo.py", line 327, in _cron
    self._scheduler(repos.ids)
  File "/srv/openerp/instances/runbot2-1/parts/runbot/runbot/models/repo.py", line 256, in _scheduler
    build_ids._schedule()
  File "/srv/openerp/instances/runbot2-1/parts/runbot-addons/runbot_travis2docker/models/runbot_build.py", line 200, in _schedule
    res = super(RunbotBuild, self)._schedule()
  File "/srv/openerp/instances/runbot2-1/parts/runbot/runbot/models/build.py", line 463, in _schedule
    build._schedule()
  File "/srv/openerp/instances/runbot2-1/parts/runbot-addons/runbot_travis2docker/models/runbot_build.py", line 200, in _schedule
    res = super(RunbotBuild, self)._schedule()
  File "/srv/openerp/instances/runbot2-1/parts/runbot/runbot/models/build.py", line 463, in _schedule
    build._schedule()
  File "/srv/openerp/instances/runbot2-1/parts/runbot-addons/runbot_travis2docker/models/runbot_build.py", line 211, in _schedule
    ssh_keys = build._get_ssh_keys() or ''
  File "/srv/openerp/instances/runbot2-1/parts/runbot-addons/runbot_travis2docker/models/runbot_build.py", line 195, in _get_ssh_keys
    "Error fetching %s (%s): %s", own_key, login, err)
ValueError: <class 'UnboundLocalError'>: "local variable 'login' referenced before assignment" while evaluating
'model._cron()'
```

This points to the `login = ''` being necessary.

Reply:
But I agree it is strange to land there, because the 'if not response: return' lines above should have caught this.